llvm/tools/llvm-pdbutil/llvm-pdbutil.cpp

```diff
@@ -880,6 +883,23 @@ static void yamlToPdb(StringRef Path) {
     IpiBuilder.addTypeRecord(Type.RecordData, std::nullopt);
   }
 
+  auto &GsiBuilder = Builder.getGsiBuilder();
+  const auto &Publics = YamlObj.PublicsStream.value_or(DefaultPublicsStream);
```

Comment:
Maybe I'm missing something here, but doesn't the `const auto&` bind to a temporary? Because `YamlObj.PublicsStream` is a `std::optional<PdbPublicsStream>`, then the return of `value_or` is a `PdbPublicsStream`.

The other streams above also do this, and I'd guess this would break in some way if it would bind to a temporary.

Reply:
Nevermind, it's lifetime extended.